src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvThread.cs

```diff
@@ -112,7 +112,7 @@ public async Task StopAsync(TimeSpan timeout)
                 Post(t => t.AllowStop());
                 if (!await WaitAsync(_threadTcs.Task, stepTimeout).ConfigureAwait(false))
                 {
-                    _log.LogCritical($"{nameof(LibuvThread)}.{nameof(StopAsync)} failed to terminate libuv thread, {nameof(AllowStop)}");
+                    _log.LogWarning($"{nameof(LibuvThread)}.{nameof(StopAsync)} failed to terminate libuv thread, {nameof(AllowStop)}");
```

Comment:
This kinda feels like a cheap way to avoid reverting your change rather than finding the real regression...

Reply:
My bad. I thought this was removing the line on 134. I need to pay more attention. Carry on.

Reply:
This is pretty rare, right? We think this is due to a thread starved environment?

Reply:
That's my current guess.